feature/policy_forwarding/otg_tests/mpls_gre_ipv4_encap_scale_test/mpls_gre_ipv4_encap_scale_test.go

```diff
@@ -0,0 +1,566 @@
+func GetDefaultOcPolicyForwardingParams() cfgplugins.OcPolicyForwardingParams {
+	return cfgplugins.OcPolicyForwardingParams{
+		NetworkInstanceName: "DEFAULT",
+		InterfaceID:         "Agg1.10",
+		AppliedPolicyName:   "customer1",
+	}
+}
+
+func configureInterfacePropertiesScale(t *testing.T, dut *ondatra.DUTDevice, aggID string, ocPFParams cfgplugins.OcPolicyForwardingParams, interfaces []*attrs.Attributes) {
+	_, _, pf := cfgplugins.SetupPolicyForwardingInfraOC(ocPFParams.NetworkInstanceName)
+
+	cfgplugins.InterfacelocalProxyConfigScale(t, dut, interfaces, aggID)
+	cfgplugins.InterfaceQosClassificationConfigScale(t, dut, interfaces, aggID)
+	cfgplugins.InterfacePolicyForwardingConfigScale(t, dut, interfaces, aggID, pf, ocPFParams)
+}
+
+// function should also include the OC config , within these deviations there should be a switch statement is needed
+// Modified to accept pf, ni, and ocPFParams
+func EncapMPLSInGRE(t *testing.T, dut *ondatra.DUTDevice, pf *oc.NetworkInstance_PolicyForwarding, ni *oc.NetworkInstance, encapParams *EncapParams, ocPFParams cfgplugins.OcPolicyForwardingParams, ocNHGParams cfgplugins.StaticNextHopGroupParams) {
+	cfgplugins.MplsConfig(t, dut)
+	cfgplugins.QosClassificationConfig(t, dut)
+	cfgplugins.LabelRangeConfig(t, dut)
+	cfgplugins.NextHopGroupConfigScale(t, dut, encapParams.intCount, encapParams.mplsStaticLabels, encapParams.mplsStaticLabelsForIpv6, ni, ocNHGParams)
+	cfgplugins.PolicyForwardingConfigScale(t, dut, encapParams.intCount, pf, ocPFParams)
+	if !deviations.PolicyForwardingOCUnsupported(dut) {
+		PushPolicyForwardingConfig(t, dut, ni)
+	}
+}
+
+// TestMPLSOGREEncapIPv4 tests MPLSOGRE Encap test case for IPv4 flow.
+func TestMPLSOGREEncapIPv4(t *testing.T) {
+	ate := ondatra.ATE(t, "ate")
+	t.Log("PF-1.15.2: Verify PF MPLSoGRE encapsulate action for IPv4/IPv6 traffic")
+	createflow(t, top, FlowIPv4, true)
+	createflow(t, top, FlowIPv6, false)
+	sendTraffic(t, ate)
+
+	if err := FlowIPv4Validation.ValidateLossOnFlows(t, ate); err != nil {
+		t.Errorf("Validation on flows failed (): %q", err)
+	}
+	if err := FlowIPv6Validation.ValidateLossOnFlows(t, ate); err != nil {
+		t.Errorf("Validation on flows failed (): %q", err)
+	}
+	FlowIPv4.VLANFlow.VLANId = 20
+	FlowIPv4.VLANFlow.VLANCount = 0
+	FlowIPv4.IPv4Flow.RawPriority = 1
+	FlowIPv4.IPv4Flow.RawPriorityCount = 0
+	FlowIPv4.PacketsToSend = 1000
+
+	createflow(t, top, FlowIPv4, true)
+	packetvalidationhelpers.ConfigurePacketCapture(t, top, EncapPacketValidation)
+	sendTrafficCapture(t, ate)
+	if err := FlowIPv4Validation.ValidateLossOnFlows(t, ate); err != nil {
+		packetvalidationhelpers.ClearCapture(t, top, ate)
+		t.Errorf("Validation on flows failed (): %q", err)
+	}
+
+	if err := packetvalidationhelpers.CaptureAndValidatePackets(t, ate, EncapPacketValidation); err != nil {
+		packetvalidationhelpers.ClearCapture(t, top, ate)
+		t.Errorf("Capture And ValidatePackets Failed (): %q", err)
+	}
+	packetvalidationhelpers.ClearCapture(t, top, ate)
```

Comment:
https://go.dev/doc/effective_go#defer

```suggestion
	defer packetvalidationhelpers.ClearCapture(t, top, ate)
	if err := FlowIPv4Validation.ValidateLossOnFlows(t, ate); err != nil {
		t.Errorf("Validation on flows failed (): %q", err)
	}

	if err := packetvalidationhelpers.CaptureAndValidatePackets(t, ate, EncapPacketValidation); err != nil {
		t.Errorf("Capture And ValidatePackets Failed (): %q", err)
	}
```